src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2125,38 +2127,119 @@ void CodeGen::genFMAIntrinsic(GenTreeHWIntrinsic* node)
     regNumber op1Reg;
     regNumber op2Reg;
 
+    regNumber op1RegOld = op1->GetRegNum();
+    regNumber op2RegOld = op2->GetRegNum();
+    regNumber op3RegOld = op3->GetRegNum();
+
+    // For special case that targetReg equals to more than one op Regs
+    bool sameReg = false;
+    if ((op1RegOld == op2RegOld && targetReg == op1RegOld) || (op1RegOld == op3RegOld && targetReg == op1RegOld) ||
```

Comment:
You can simplify this with:

```c++
uniqueOverwrittenReg = true;
if ((targetReg == op1RegOld) && ((op1RegOld == op2RegOld) || (op2RegOld == op3RegOld) || (op1RegOld == op3RegOld))
{
   uniqueOverwrittenReg = false;
}
```

Also rename `sameReg` to something meaningful like `uniqueOverwrittenReg`.

Reply:
@kunalspathak, there was a bug in my code, the third condition should be ```op2RegOld == op3RegOld && targetReg == op2Reg```. Therefore we could not simplify this. Sorry for the confusion!

Reply:
Ah, that's true...it should be matching one of the same operandReg.